ansible_wisdom/ai/api/formatter.py

```diff
@@ -353,15 +353,31 @@ def get_task_names_from_tasks(tasks):
     return names
 
 
-def restore_original_task_names(output_yaml, prompt):
+def restore_original_task_names(output_yaml, prompt, payload_context=''):
     if output_yaml and is_multi_task_prompt(prompt):
+        full_yaml = payload_context + output_yaml
+        try:
+            data = yaml.safe_load(full_yaml)
+        except Exception as exc:
+            logger.exception(f'Error while loading the result role/playbook YAML:{exc}')
+            return
```

Comment:
@robinbobbitt what do you mean by returning `None`? for the `payload_context` or the `output_yaml`? By `the calling code` you mean the `data = yaml.safe_load(full_yaml)` line? 

This part of the code just tries to be safe with the YAML stuff. if there is a problem with the YAML content, then we return and the `restore_original_task_names` simply do nothing.

Reply:
@mabulgu I'm talking about line 363 where we do a `return` with no value, which returns `None`. Re: calling code, I'm referring to the code that calls `restore_original_task_names`.

Reply:
I think the current place of the `restore_original_task_names` is good because we pass the`anonymized_recommendation_yaml` and get the `recommendation_yaml` from the function. The yaml that I want to check in the `restore_original_task_names` is the full yaml so it is different in my case. I dont check the recommendation yaml only but the whole yaml with the potential payload context  if it is a playbook. 

Your idea of returning `output_yaml` makes alot sensein this case because if we have a problem with restoring the original task names because of a problem with the merged yaml (`payload_context + output_yaml`) then we should have the raw, unchanged one as the returned value to continue our validation like `yaml.safe_load(recommendation_yaml)` in the `post_process.py`. I've made the changes reg. this.